domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -1960,19 +1982,19 @@ private void closeIfNonNull(Transaction transaction) {
         }
         resultsHelper.offset++;
       } catch (Exception e) {
+        streamResult.handler(null);
         log.error(e.getMessage(), e);
         if (!promise.future().isComplete()) {
           promise.complete(streamResult);
           replyHandler.handle(promise.future());
         }
         rowStream.close();
-        closeIfNonNull(transaction);
-        streamResult.fireExceptionHandler(e);
+        closeIfNonNull(transaction).onComplete((AsyncResult<Void> voidRes) -> streamResult.fireExceptionHandler(e));
       }
     }).endHandler(v2 -> {
       rowStream.close();
-      closeIfNonNull(transaction);
-      resultInfo.setTotalRecords(
+      closeIfNonNull(transaction).onComplete((AsyncResult<Void> voidRes) -> {
```

Comment:
```suggestion
      closeIfNonNull(transaction).onComplete(ignore -> {
```
Braces not needed. Type declaration not needed.
The name `ignore` makes clear that we ignore a close failure on purpose.